File: flink-test/flink-dim-java/src/main/java/com.flink.dim/externalconnectors/DBConnection.java
package com.flink.dim.externalconnectors;

import org.apache.flink.types.Row;
import scala.collection.Iterator;

import java.io.Serializable;

/**
 * 应用模块名称.
 * <p>
 * 代码描述
 * </P>
 *
 * @author xx
 * @since 2020/3/24 14:04
 */
public interface DBConnection extends Serializable {
    public void open();

    public Iterator<Row> batchQuery();

    public Row query(Row q);

    public Boolean release();
}

File: flink-test/flink-dim/src/main/scala/com/flink/dim/externalconnectors/DBConnection.scala
package com.flink.dim.externalconnectors

/**
  * Connector for dimension table queries.
  *
  */
trait DBConnection extends Serializable{

  import org.apache.flink.types.Row

  /**
    * Initialize connection .
    *
    */
  def open
  /**
    * Batch query dimension table.
    * @throws DimensionJoinException Query Exception
    * @return The result set of a dimension table query
    */
  @throws[DimensionJoinException]
  def batchQuery(): Iterator[Row]
  /**
    * Query dimension table one by one.
    *
    * @throws DimensionJoinException Query Exception
    * @param q Query condition of one by one
    * @return The result of a dimension table query
    */
  @throws[DimensionJoinException]
  def query(q: Row): Row
  /**
    * Release resource.
    *
    */
  def release: Boolean
}
